### PR TITLE
Dippable Component v2 - No Extraneous Commits Edition

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
:cl: Y0SH1_M4S73R
add: Food and sharp objects can be dipped in reagents. Sharp dipped objects will transfer some of their reagents to their victims via touch when attacking.
/:cl:

This adds the "dippable" component, whose parameters are explained in the comments at the top of dippable.dm.

Hopefully this time there won't be any extraneous commits crammed into this PR. Like before, feel free to address any balance concerns, and in this case, any rebase blunders.

### CODE REVIEW

**Comment by praisenarsie** [comment]
again, blame cyberboss for those commits

**Comment by davethwave** [comment]
New zombie virus cure is glass shard dipped in green slime juice

**Comment by zamolxius** [comment]
NICE!

**Comment by kevinz000** [comment]
so how do sharp objects work do they inject on hit? because that doesn't seem very  balanced to me.

**Comment by Fox-McCloud** [comment]
This isn't a good idea.

I've seen this on Goonstation; it was supported there, very strongly---and to be fair, it's a neat feature, but the potential for abuse is massive.

It's basically giving anything that's sharp parapen status with a bit less stealth.

This is going to turn weapons like the armblade, saw, etc from high damage weapons to high damage *instant-stun* weapons or ones that also deal copious amounts of damage over time. 

I've seen this play out on Goon; it was a disaster.

**Comment by ExcessiveUseOfCobblestone** [comment]
INJECTION SPECIFIC EFFECTS OR BUST

**Comment by kevinz000** [comment]
yeah ok your average kitchen knife being a hypospray is dumb.

**Comment by kevinz000** [comment]
you forget we have things like neurotoxin in the codebase.

**Comment by 81Denton** [comment]
Like kevin and fox said, this will fuck over weapons balancing as soon as you dip your weapon into any of the stun chems, or even the racial mutagens as those drop you to the floor. Or even any of the 50 fast-acting poisons out there.

**Comment by arsserpentarium** [comment]
Well if you plan knife fight,why not to dip knife into neurotoxin,or at least mud from teeth or shit if you haven't neurotoxin.

**Comment by 81Denton** [comment]
@arsserpentarium That change would turn any spear/knife into a stunbaton or worse. Most strong toxins and stun chems work at low dosages; my best guess is that they're based around syringes.

**Comment by davethwave** [comment]
Just have the injection rate of any liquid you dip on something .5 or something and a waste value of 3 or so so if you dip a weapon is 15 units of liquid after every hit would use 3.5 or so units. .5 inside the person and the other 3 are gone due to ect reasons and since things tend to dry in the air any weapon you hold that's in contact to air would start to lose .2 units a second of what you dipped it in or something like that.

**Comment by 81Denton** [comment]
@Y0SH1M4S73R Regarding balancing:

- Under 5u of Neurotoxin stun someone long enough to cuff
- Under 10u of proper toxins (Cyanide, Curare, Lexorin, Coniine, Amanitin, Pancuronium, Initropidril and many more) will kill players that don't receive medical attention within one minute.
- Or you can just dip it in facid etc. for straight up damage.

tl;dr dipping will either barely do anything or be broken and require a rework of a lot of chems. Syringe guns are at least balanced around the low amount of shots, slow speed and the inability to pierce hardsuits/firesuits etc.

**Comment by Y0SH1M4S73R** [comment]
To everyone concerned about ghetto hyposprays, I can drastically reduce the amount of chems that dipped items can contain and drastically reduce the efficiency of the transferrence of dipped chems. Barring that, I can completely remove the ability to dip sharp items and leave the component in for cigarettes and foodstuffs.

**Comment by kevinz000** [comment]
@Y0SH1M4S73R you don't understand
anything that blocks syringes will likely be a slowdown and something that most people won't wear.
even injecting 0.5 units is an instant death if it's something like neurotoxin.
it doesn't matter how much you inject if it injects at all it's GG because chems work at extremely low doses. 
@arsserpentarium because not everyone has access to neurotoxin and it being a requirement to have ak nife fight (which the point is partially not have one hit kill like stunbaton fights) is dumb.

**Comment by 81Denton** [comment]
@Y0SH1M4S73R It's honestly a pain in the dong to balance, but hats off to coding all this.

**Comment by kevinz000** [comment]
agreed. it's a good idea and it sounds fun trust me but it will get abused to the point of not being in actuality fun.

**Comment by arsserpentarium** [comment]
But you'll need to find robust reagent.
Also different things can have different amount of reagents to hold and transfer rates. Also reagent can vanish over time from object.

**Comment by Rinoka** [comment]
".5 of neurotoxin will kill"

this sounds like a problem with how ridiculously OP chemistry is in current build, _not_ a problem with this PR. as it is, botanists can make extremely similar things with very little effort, as long as they can get mutagen. chems are extremely potent, not even mentioning when you stack them.

it doesn't really make sense for them to have such ridiculous numbers that .x of a toxin could kill.

personally i'd rather see this be implemented and then the more ridiculous chems get their damage/tick rate reduced as is needed to maintain balance. right now, this would add interesting things to the meta; i'm all for it.

**Comment by arsserpentarium** [comment]
I need to remind you. Balance!=nerf.
Yes,tgchem is robust. But that's one of reasons,why we love it and play on tg, not bay. Where nerfing even spoons.

**Comment by 81Denton** [comment]
@Rinoka That would mean that someone has to rebalance almost every single chemical in game, just so they don't become OP when dipped. Do you have someone in mind who's willing to do that fuckton of work?

**Comment by Rinoka** [comment]
@81Denton really that's been coming for a while, small-u dosages shouldn't be downright murderous, so while i can't think of anyone, i'm already planning on doing at least a spreadsheet for data visualization, to figure out which chems can literally kill with <5u unassisted.

@arsserpentarium there's a difference between robust and "a bit too fucking much". if you have access to most toxin chems, there shouldn't be any where you're able to kill people with 1u, because you should be able to administer things in high-u doses. (besides, one of the more robust things about our system is that you can load multiple poisons into one dosage, and have them all deal damage at once - which is realistic, but means that individual chems being powerful enough to kill with .5u means you can kill _extremely fast._)

**Comment by ShizCalev** [comment]
conflict.

**Comment by Cyberboss** [comment]
I think I speak for @KorPhaeron when I say we don't want to add more hypos
